File: MiniDOMBase/DMD/Collections/CSyncKeyCollectionOfT.cs
using System;

namespace minidom
{
    [Serializable]
    public class CSyncKeyCollection<T> : CSyncKeyCollection
    {
        public CSyncKeyCollection()
        {
        }

        public CSyncKeyCollection(CKeyCollection col) : base(col)
        {
        }

        public new T this[int index]
        {
            get
            {
                return (T)base[index];
            }

            set
            {
                base[index] = value;
            }
        }

        public new T this[string key]
        {
            get
            {
                return (T)base[key];
            }

            set
            {
                base[key] = value;
            }
        }
    }
}

File: MiniDOMBase/DMD/Collections/CSyncKeyCollectionOfT.vb
Imports minidom.Sistema

<Serializable>
Public Class CSyncKeyCollection(Of T)
    Inherits CSyncKeyCollection

    Public Sub New()
    End Sub

    Public Sub New(ByVal col As CKeyCollection)
        MyBase.New(col)
    End Sub

    Default Public Shadows Property Item(ByVal index As Integer) As T
        Get
            Return MyBase.Item(index)
        End Get
        Set(value As T)
            MyBase.Item(index) = value
        End Set
    End Property

    Default Public Shadows Property Item(ByVal key As String) As T
        Get
            Return MyBase.Item(key)
        End Get
        Set(value As T)
            MyBase.Item(key) = value
        End Set
    End Property

End Class
